core/lib/src/unix.rs

```diff
@@ -4,11 +4,28 @@ use rocket_http::hyper::net::NetworkStream;
 #[cfg(all(unix, feature = "tls"))] use rocket_http::unix::UnixHttpsStream;
 use data::NetStream;
 
+macro_rules! check_unix_domain_socket {
+    ($path:expr) => (
+        use std::fs;
+        use fs2::FileExt;
+
+        let lock_path = $path.with_extension("lock");
+        let lock_file = fs::OpenOptions::new().read(true).write(true).create(true).open(lock_path).unwrap();
+        if let Err(err) = lock_file.try_lock_exclusive() {
+            return $crate::error::LaunchError::new($crate::error::LaunchErrorKind::Io(err));
+        }
+        if $path.exists() {
+            fs::remove_file($path).unwrap();
```

Comment:
Same here with returning a `LaunchError`.

Explicitly checking for existence before removal is normally a race condition, but that's probably okay here since it's inside the lock and avoids some extra error finagling afterwards.